clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -82,6 +87,96 @@
 
+    public class Builder {
+        private final String name;
+        private final Type type;
+        private final Importance importance;
+        private String documentation = "";
+        private Validator validator = null;
+        private Object defaultValue = NO_DEFAULT_VALUE;
+        private String group = null;
+        private int orderInGroup = -1;
+        private Width width = Width.NONE;
+        private String displayName;
+        private List<String> dependents = Collections.EMPTY_LIST;
+        private Recommender recommender = null;
+        private boolean internalConfig = false;
+        private Collection<String> deprecatedNames = Collections.EMPTY_SET;
+
+        public Builder(String name, Type type, Importance importance) {
+            this.name = name;
+            this.type = type;
+            this.importance = importance;
+            this.displayName = name;
+        }
+
+        public Builder documentation(String documentation) {
+            this.documentation = documentation;
+            return this;
+        }
+
+        public Builder validator(Validator validator) {
+            this.validator = validator;
+            return this;
+        }
+
+        public Builder defaultValue(Object defaultValue) {
+            this.defaultValue = defaultValue;
+            return this;
+        }
+
+        public Builder group(String group) {
+            this.group = group;
+            return this;
+        }
+
+        public Builder orderInGroup(int orderInGroup) {
+            this.orderInGroup = orderInGroup;
+            return this;
+        }
+
+        public Builder width(Width width) {
+            this.width = width;
+            return this;
+        }
+
+        public Builder displayName(String displayName) {
+            this.displayName = displayName;
+            return this;
+        }
+
+        public Builder dependents(List<String> dependents) {
+            this.dependents = dependents;
+            return this;
+        }
+
+        public Builder recommender(Recommender recommender) {
+            this.recommender = recommender;
+            return this;
+        }
+
+        public Builder internalConfig(boolean internalConfig) {
+            this.internalConfig = internalConfig;
+            return this;
+        }
+
+        public Builder deprecatedNames(Collection<String> deprecatedNames) {
+            this.deprecatedNames = deprecatedNames;
+            return this;
+        }
+
+        public ConfigDef build() {
+            ConfigKey configKey = new ConfigKey(name, type, defaultValue, validator, importance,
+                documentation, group, orderInGroup, width, displayName, dependents, recommender,
+                internalConfig, deprecatedNames);
+            return define(configKey);
+        }
+    }
+
+    Builder buildDefinition(String name, Type type, Importance importance) {
+        return new Builder(name, type, importance);
```

Comment:
Should this be simply `builder`? The current name sounds like it builds the definition.

Reply:
How about `configBuilder`